### PR TITLE
Update GPU details for ctc_loss

### LINKED ISSUE #55290: tf.nn.ctc_loss documentation misleads about GPU support, leading to 50x performance difference
## URL(s) with the issue:

https://www.tensorflow.org/api_docs/python/tf/nn/ctc_loss

## Description of issue (what needs changing):

[The documentation for `tf.nn.ctc_loss`](https://github.com/tensorflow/tensorflow/blob/v2.8.0/tensorflow/python/ops/ctc_ops.py#L790) states:
> On TPU and GPU: Only dense padded labels are supported.
> On CPU: Caller may use SparseTensor or dense padded labels but calling with a SparseTensor will be significantly faster.

However, reading through [the code for the CTC loss implementation](https://github.com/tensorflow/tensorflow/blob/v2.8.0/tensorflow/python/ops/ctc_ops.py#L921-L957) shows that SparseTensors _are_ supported on GPU, and in practice, the SparseTensor code path uses CUDNN and is about 50x faster than the non-sparse code path.

I have tested this on GPU, but not on TPU, where I expect the documentation may be accurate (as the CUDNN code path doesn't apply to TPU). 

### Submit a pull request?

Would be happy to, after confirmation that this is indeed an issue.

cc @kaixih, @f90

### PR DESCRIPTION
**Pull request by sachinprasadhs**
Updated GPU details for ctc_loss which supports dense padded labels and SparseTensor.
Closes #55290

### CODE REVIEW

**Comment by f90** [comment]
Hey, thanks for looking into this. Can we add to the docs here a note that the code paths for sparse labels and dense labels are different on GPU? I.e. if you provide sparse labels, you are using the cuDNN implementation and you can be 50x faster than with dense labels. People will otherwise keep using dense labels and have no way of knowing why the loss is so slow even with a GPU

**Comment by gbaned** [comment]
@Rohan Jain Can you please review this PR ? Thank you!

**Comment by gbaned** [comment]
@rohan100jain Can you please review this PR ? Thank you!

File: tensorflow/python/ops/ctc_ops.py [hunk 1]
```diff
@@ -897,7 +897,7 @@ def ctc_loss_v3(labels,
     setting of preprocess_collapse_repeated=False, ctc_merge_repeated=True
   - Labels may be supplied as either a dense, zero-padded tensor with a
     vector of label sequence lengths OR as a SparseTensor.
-  - On TPU and GPU: Only dense padded labels are supported.
+  - On TPU: Only dense padded labels are supported.
   - On CPU: Caller may use SparseTensor or dense padded labels but calling with
```
Comment (cantonios) [inline tensorflow/python/ops/ctc_ops.py:901]: Change this to "On CPU and GPU:  Caller may..."

since (as noted by other commenters) using sparse labels will be significantly faster.

**Review by cantonios — CHANGES_REQUESTED** [review]

**Review by cantonios — APPROVED** [review]
